daemon/start.go

```diff
@@ -64,10 +61,7 @@ func (daemon *Daemon) setHostConfig(container *Container, hostConfig *runconfig.
 		// ensure the source exists on the host
 		_, err := os.Stat(source)
 		if err != nil && os.IsNotExist(err) {
-			err = os.MkdirAll(source, 0755)
-			if err != nil {
-				return fmt.Errorf("Could not create local directory '%s' for bind mount: %v!", source, err)
-			}
+			return fmt.Errorf("Source '%s' does not exist !", source);
```

Comment:
I don't think we want to return an error here, just remove this whole for loop since it (should) be handled by the volume subsystem (something the integration tests should find out quickly enough)